core/src/consensus/tendermint/engine.rs

```diff
@@ -165,11 +173,40 @@ impl ConsensusEngine for Tendermint {
             }
             (header.number(), term_id)
         };
+        let rewards = stake::drain_previous_rewards(&mut block.state_mut())?;
+        let client = self
+            .client
+            .read()
+            .as_ref()
+            .ok_or(EngineError::CannotOpenBlock)?
+            .upgrade()
+            .ok_or(EngineError::CannotOpenBlock)?;
+
+        let (start_of_the_next_term, start_of_the_current_term) = {
```

Comment:
IMO, 
`start_of_the_current_term == end_of_the_previous_term + 1`

Reply:
This line should be `let (start_of_the_current_term, end_of_the_prev_prev_term) = {`